stampifier.py

```diff
@@ -1,32 +1,75 @@
-''' This module contains the main interface definition
-for the classifier and summarizer module
-'''
+"""This script creates Stampify"""
+
+import logging
+
 from classification.classifier_main import Classifier
+from data_models.website import Website
+from error import InvalidUrlError, WebsiteNotStampifiableError
+from extraction.extractor import Extractor
+from stamp_generation.stamp_generator import StampGenerator
 from summarization.extractor_output_preprocessor import \
     ExtractorOutputPreprocessor
 from summarization.summarizer_main import Summarizer
 
+LOGGER = logging.getLogger()
+LOG_FILENAME = 'website.log'
+logging.basicConfig(filename=LOG_FILENAME, level=logging.DEBUG)
 
-class ClassifierAndSummarizer:
-    def __init__(self, contents, max_pages):
-        '''
-        Params:
-          * contents : an object of type contents
-            it contains the contents_list
-          * max_pages : the maximum pages required
-        '''
-        self.contents = contents
+
+class Stampifier:
+
+    def __init__(self, url, max_pages):
+        self.url = url
         self.max_pages = max_pages
+        self._website = Website(self.url)
         self.is_stampifiable = False
         # object of type StampPages
         self.stampified_pages = None
 
+    def stampify(self):
+        """Starts the stampification process"""
+
+        if not self._website.is_valid:
+            raise InvalidUrlError(self._website.url)
+
+        return self.__convert_website_to_stamp()
+
+    def __convert_website_to_stamp(self):
+        """This method is the pipeline between all the modules"""
+
+        _extractor = Extractor(self._website.url)
+        self._website.set_contents(_extractor.extract_html())
+
+        LOGGER.debug(self._website.convert_to_dict())
+
+        if not self._website.contents.content_list:
+            raise WebsiteNotStampifiableError(
+                message="No content extracted!",
+                failure_source="Extractor")
+
+        _classifier_and_summarizer_response = self.get_stampified_content()
```

Comment:
get_stampified_content should only return stamp_pages.
Since error will be raised in Classifier itself.

Reply:
@telescopic

Reply:
@telescopic Can you look into this once this PR is merged and place the exceptions accordingly?

Reply:
Noted. I'll raise issues for the same.